packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -26,6 +26,7 @@ const stripPosition = require(`unist-util-remove-position`)
 const hastReparseRaw = require(`hast-util-raw`)
 
 let fileNodes
+let tocOptionsCacheStr = ``
```

Comment:
this won't work - queries are executed in parallel - it's not safe to set global context - you would need to change `tableOfContentsCacheKey` function to accept second param (tocOptions):

```js
const tableOfContentsCacheKey = (node, tocOptions) =>
  `transformer-remark-markdown-toc-${
    node.internal.contentDigest
  }-${pluginsCacheStr}-${pathPrefixCacheStr}`
  }-${pluginsCacheStr}-${pathPrefixCacheStr}-${JSON.stringifu(tocOptions)}`
```